### PR TITLE
Fixes Revs & Clockcultists being valid for mindswap

### LINKED ISSUE #10431: Mind-swapping with a loyalty implanted person as a rev keeps your rev status while still showing you as being implanted.
Basically what the title says.
May become an issue with datum antags interacting with each other in the future.
@Incoming5643

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #10431

Also replaces a ton of datum usage with the proper defines.

🆑 ShizCalev
fix: Clockcultists and revolutionaries can no longer be mindswapped.
/🆑

### CODE REVIEW

File: code/modules/spells/spell_types/mind_transfer.dm [hunk 1]
```diff
@@ -54,7 +53,8 @@ Also, you never added distance checking after target is selected. I've went ahea
 		to_chat(user, "<span class='warning'>You're killing yourself! You can't concentrate enough to do this!</span>")
 		return
 
-	if((target.mind.special_role in protected_roles) || cmptext(copytext(target.key,1,2),"@"))
+	var/datum/mind/TM = target.mind
+	if((TM.has_antag_datum(ANTAG_DATUM_WIZARD) || TM.has_antag_datum(ANTAG_DATUM_CULT) || TM.has_antag_datum(ANTAG_DATUM_CLOCKCULT) || TM.has_antag_datum(ANTAG_DATUM_CHANGELING) || TM.has_antag_datum(ANTAG_DATUM_REV)) || cmptext(copytext(target.key,1,2),"@"))
```
Comment (ShizCalev) [inline code/modules/spells/spell_types/mind_transfer.dm]: actual change is here for skimmers.

**Comment by Buggy123** [comment]
So if you have mindswap you have a quick and 100% accurate way to tell if someone is a cultist or a rev?

**Comment by ShizCalev** [comment]
Was already that way for blood cultists and changelings, clockcultists and revs were missing from the old protected roles list. It's meant to prevent the issue linked in the op.

**Comment by Buggy123** [comment]
Fair enough.

**Comment by Cyberboss** [comment]
Conflicts

**Pull request merged**